katsdpimageutils/test/test_primary_beam_correction.py

```diff
@@ -93,19 +104,28 @@ def test_nans(self):
 
 class TestWriteNewFits:
     def setup(self):
+        # create temporary directory
+        self.tmpdir = tempfile.TemporaryDirectory()
         in_array = np.ones([1, 10, 20, 20])
         hdu = fits.PrimaryHDU(in_array)
         hdu.header['history'] = 'AIPS   CLEAN BMAJ=  5e-03 BMIN=  1e-03 BPA=  42.11'
-        if os.path.isfile('/tmp/testdir/new.fits'):
-            pass
-        else:
-            os.mkdir("/tmp/testdir")
-            hdu.writeto('/tmp/testdir/new.fits')
+        hdu.writeto(self.tmpdir.name + '/image.fits')
 
-    def tearDown(self):
-        shutil.rmtree("/tmp/testdir")
+    def teardown(self):
+        self.tmpdir.cleanup()
 
     def test_case(self):
-        out_array = 2 * np.ones([1, 1, 20, 20])
-        write_new_fits(out_array, '/tmp/testdir/new.fits', '/tmp/testdir/new_pbc.fits')
-        assert os.path.isfile('/tmp/testdir/new_pbc.fits')
+        pbc_array = 2 * np.ones([1, 1, 20, 20])
+        tmpdir = self.tmpdir
+        path = tmpdir.name + '/image.fits'
+        outFileName = tmpdir.name + '/image_pbc.fits'
```

Comment:
Rather use `os.path.join` to construct the filenames. You could also store the input filename during `setup` so that you don't need to re-compute it here.

Reply:
Noted and fixed.